app/test/testattributecontroller.cpp

```diff
@@ -718,30 +725,182 @@ void TestAttributeController::testPhotoRenaming()
+  QCOMPARE( fieldToTest->name(), QStringLiteral( "photo_with_slash_in_subfolder_relative_to_project_path" ) );
+
+  originalPath = projectDir + QStringLiteral( "/media/with_slash/image5.jpg" );
+  expectedNewPath = projectDir + QStringLiteral( "/media/with_slash/custom_photo_name_format/Survey_1.jpg" );
+  expectedNewFieldValue = QStringLiteral( "media/with_slash/custom_photo_name_format/Survey_1.jpg" );
+
+  testcases << testcase{ originalPath, expectedNewPath, fieldIdx, expectedNewFieldValue };
+
+  //
+  // Field "photo_with_slash_in_subfolder_relative_to_default_path"
+  //  - Contains custom photo name expression with subfolder 'custom_photo_name_format'
+  //  - Images should be stored inside folder 'media/with_slash' (via default path in the field config)
+  //  - Stores paths relative to default path (folder should not be part of the field value)
+  //
+  //  - this is kind-of project misconfiguration, the widget is configured to store photos
+  //    in 'media/with_slash', but that is overriden by the custom photo name as it includes
+  //    a subfolder and the project is configured to store paths relative to default path.
+  //    The value in the field will include the photo name format, even though the widget
+  //    will look for images inside 'media/with_slash' folder ... plugin should report this.
+  //
+
+  fieldIdx = 8;
+  fieldToTest = controller.formItem( items.at( fieldIdx ) );
+  QCOMPARE( fieldToTest->name(), QStringLiteral( "photo_with_slash_in_subfolder_relative_to_default_path" ) );
+
+  originalPath = projectDir + QStringLiteral( "/media/with_slash/image6.jpg" );
+  expectedNewPath = projectDir + QStringLiteral( "/media/with_slash/custom_photo_name_format/Survey_2.jpg" );
+  expectedNewFieldValue = QStringLiteral( "custom_photo_name_format/Survey_2.jpg" );
+
+  testcases << testcase{ originalPath, expectedNewPath, fieldIdx, expectedNewFieldValue };
+
+  const QgsFeature &f = controller.featureLayerPair().featureRef();
+
+  for ( const auto &testcase : std::as_const( testcases ) )
+  {
+    qInfo() << QStringLiteral( "Testing field #%1" ).arg( testcase.fieldIdx );
+
+    QVERIFY( !QFile::exists( testcase.originalPath ) );
+    QVERIFY( QFile::exists( testcase.expectedNewPath ) );
+
+    QCOMPARE( f.attribute( testcase.fieldIdx ), testcase.expectedNewFieldValue );
+  }
 
-  QVERIFY( !QFile::exists( projectDir + QStringLiteral( "/photo2.jpg" ) ) );
-  QVERIFY( QFile::exists( projectDir + QStringLiteral( "/photos/Survey.jpg" ) ) );
-  QCOMPARE( controller.featureLayerPair().feature().attribute( 4 ), QVariant( "photos/Survey.jpg" ) );
+  // TODOs:
+  //  - report that people should not use feature IDs in custom photo names!
+  //  - fix issue where email might be used for mm_username variable as this breaks the renaming!
+  //  - fix docs recommending variables with old prefix for photo names
+  //  - report misconfiguration of the use of subfolders & subfolders in the photo naming
```

Comment:
I think we can remove this now

Reply:
Or move them to plugin repo :)

Is the email a real issue though? Emails should not have illegal characters for a filename!